wallet/src/de/schildbach/wallet/ui/TransactionsAdapter.java

```diff
@@ -514,17 +515,26 @@ private void bind(final Transaction tx) {
 
             // fiat value
             final ExchangeRate exchangeRate = tx.getExchangeRate();
+            fiatView.setAmount(null);  //clear the exchange rate first
             if (exchangeRate != null) {
                 fiatView.setFormat(Constants.LOCAL_FORMAT.code(0,
                         PREFIX_ALMOST_EQUAL_TO + exchangeRate.fiat.getCurrencyCode() + " "));
                 Coin absCoin = Coin.valueOf(Math.abs(txCache.value.value));
                 fiatView.setAmount(exchangeRate.coinToFiat(absCoin));
+            } else {
+                fiatView.setText(PREFIX_ALMOST_EQUAL_TO + WalletApplication.getInstance().getConfiguration().getExchangeCurrencyCode() + " ----");
             }
 
```

Comment:
Above:  set the amount to null to clear previous values -- this may not be necessary because we will set a value regardless of the value of exchangeRate (null or not null).